Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,171 @@
+- Service subnet is assumed to be 11.0.0.0/16 and is hardcoded throughout this repo. Changes to this value are not covered.
+
+A few prerequisite definitions and requirements for networking:
+
+- The **external network** is the physical network across which nodes communicate. This exists regardless of whether or not you follow this guide.
+- The **cluster subnet** is a (<a href="#allow-routing">routable</a>) virtual network that must be a /16. Each _node_ will grab a /24 from the subnet to use for its pods.
+- The **service subnet** is a hardcoded 11.0/16 network that is translated into cluster space by `kube-proxy` running on the node.
+
+**Note**: The guide will assume a local working directory of `~/kube` for the Kubernetes setup. If you choose a different directory, just replace any references to that path.
+
+## Building the Binaries ##
+We will begin with a Linux master node on a recent Ubuntu-like distro and a Windows worker node on RS3+; all of these are assumed to be VMs.
+
+First, install all of the pre-requisites on the master:
+
+    $ sudo apt-get install git build-essential docker.io conntrack
+
+### Building Kubernetes ###
+We will need to build the `kubelet` and `kubeproxy` binaries for Windows from scratch _cross-compiling from Linux_, so is a non-issue by leveraging the [standard containerized build scripts](https://github.com/kubernetes/kubernetes/tree/master/build#key-scripts) leveraged in the Kubernetes project.
+
+Various permission denied errors are avoided by building the Linux kubelet first, per the note at [acs-engine](https://github.com/Azure/acs-engine/blob/master/scripts/build-windows-k8s.sh#L176):
+
+> Due to what appears to be a bug in the Kubernetes Windows build system, one has to first build a Linux binary to generate `_output/bin/deepcopy-gen`. Building to Windows w/o doing this will generate an empty `deepcopy-gen`.
+
+You can optionally execute this in a Vagrant box:
+
+```bash
+DIST_DIR="${HOME}/kube/"
+mkdir ${DIST_DIR}
+SRC_DIR="${HOME}/src/kubernetes/"
+mkdir -p "${SRC_DIR}"
+git clone https://github.com/madhanrm/kubernetes.git ${SRC_DIR}
+cd ${SRC_DIR}
+git checkout cniwindows
+build/run.sh make WHAT=cmd/kubelet KUBE_BUILD_PLATFORMS=linux/amd64
+build/run.sh make WHAT=cmd/kubelet KUBE_BUILD_PLATFORMS=windows/amd64
+cp ${SRC_DIR}/_output/dockerized/bin/windows/amd64/kubelet.exe ${DIST_DIR}
+#winkernelproxy has already been merged but still needs to be built
+git checkout winkernelproxy 
+build/run.sh make WHAT=cmd/kube-proxy KUBE_BUILD_PLATFORMS=windows/amd64
+cp ${SRC_DIR}/_output/dockerized/bin/windows/amd64/kube-proxy.exe ${DIST_DIR}
+ls ${DIST_DIR}
+```
+
+Done! Now, we also need the actual Linux Kubernetes binaries for v1.8. You can pull these from [the Kubernetes mainline](https://github.com/kubernetes/kubernetes/releases/tag/v1.8.0), or build them from source as above, except using `K8SREPO=k8s.io/kubernetes` and the `release-1.8` branch, stopping before the `export ...` line.

```

Comment:
What 'export ..." line?

Reply:
Maybe this block needs to be written now build steps have been tweaked. I don't know what K8REPO= references now.

Reply:
Yeah this was badly merged by me. I'm fixing it. :)